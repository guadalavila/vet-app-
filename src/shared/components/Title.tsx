import React from 'react';
import { StyleSheet } from 'react-native';
import { typography } from '~shared/utils/typography';
import { size } from '~shared/utils/size';
import CustomText from './CustomText';

interface ITitleProps {
    text: string;
}

const Title = ({ text }: ITitleProps) => {
    return <CustomText style={[styles.text]}>{text}</CustomText>;
};

export default Title;

const styles = StyleSheet.create({
    text: {
        fontWeight: '600',
        fontSize: typography.size.M,
        marginLeft: size.XXL,
        marginVertical: size.XXL,
    },
});
