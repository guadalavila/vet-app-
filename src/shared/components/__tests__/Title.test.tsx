import React from 'react';
import { render, screen } from '@testing-library/react-native';
import Title from '../Title';

describe('Title', () => {
    test('should match snapshot', () => {
        const title = render(<Title text='Hello' />);
        expect(title).toMatchSnapshot();
    });
    test('should render component', () => {
        render(<Title text='Hello' />);
        expect(screen.getByText('Hello')).toBeTruthy();
    });
});
