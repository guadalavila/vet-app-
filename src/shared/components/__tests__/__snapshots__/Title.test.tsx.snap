// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should match snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#11100F",
        "fontSize": 16,
      },
      Array [
        Object {
          "fontSize": 20,
          "fontWeight": "600",
          "marginLeft": 14,
          "marginVertical": 14,
        },
      ],
    ]
  }
>
  Hello
</Text>
`;
